### PR TITLE
Add a dropdown to the heatmap to select if the heatmap should display…

### LINKED ISSUE #54: Add a dropdown to the heatmap to select if the heatmap should display: benefits, taxes, credits or net income
We still have one Heatmap but we can select the type of output,

### PR DESCRIPTION
**Pull request by ThapeloMasasa**
benefits, taxes, credits or net income drop down menu for heat map

Fixes #54

### CODE REVIEW

**Comment by ThapeloMasasa** [comment]
Here is the demo: 

https://github.com/PolicyEngine/us-marriage-incentive/assets/73863365/702c8d81-d7ed-4b57-9dd6-d63edad6ac86

**Comment by ThapeloMasasa** [comment]
https://github.com/PolicyEngine/us-marriage-incentive/assets/73863365/405b2608-3a30-43f9-9c36-c11e4faec02f

File: app.py [hunk 1]
```diff
@@ -158,14 +167,13 @@ def get_net_income(state_code, spouse=None, children_ages = {}):
     formatted_separate = list(map(lambda x: "${:,}".format(round(x)), separate))
     head_separate = programs[1]
     delta = [x - y for x, y in zip(married_programs, separate)]
-    delta_percent = [(x - y) / x if x != 0 else 0 for x, y in zip(married_programs, separate)]
+    delta_percent = [(x - y) / x if y != 0 and x != 0 else 0 for x, y in zip(married_programs, separate)]
```
Comment (PavelMakarchuk) [inline app.py:170]:
```suggestion
    delta_percent = [(x - y) / x if x != 0 and x != 0 else 0 for x, y in zip(married_programs, separate)]
```

File: app.py [hunk 2]
```diff
@@ -258,15 +262,32 @@ def get_chart():
...
         fig.update_layout(height=600, width=800)
         # Add header
         st.markdown("<h3 style='text-align: center; color: black;'>Marriage Incentive and Penalty Analysis</h3>", unsafe_allow_html=True)
         fig = format_fig(fig)
         # Display the chart
         
         st.plotly_chart(fig, use_container_width=True)
-    
-    get_chart()
+@st.cache_data()
```
Comment (PavelMakarchuk) [inline app.py:274]: need to investigate why the heatmap output differs from the table output - I believe the table is mostly aligned with the App but the heatmap seems off in all of the cases

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Comment by PavelMakarchuk** [comment]
need to fix conflicts

**Pull request merged**